src/gui/plugins/playback_scrubber/PlaybackScrubber.qml

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import QtQuick 2.9
+import QtQuick.Window 2.2
+import QtQuick.Controls 2.1
+import QtQuick.Controls.Material 2.2
+import QtQuick.Controls.Material.impl 2.2
+import QtQuick.Layouts 1.3
+import QtQuick.Controls.Styles 1.4
+
+GridLayout {
+  id: playbackScrubber
+  columns: 5
+  Layout.minimumWidth: 430
+  Layout.minimumHeight: 200
+  anchors.fill: parent
+  anchors.leftMargin: 10
+  anchors.rightMargin: 10
+  property var isPressed: false
+  property var currentTime: ""
+
+  function updateSliderValue() {
+    if (!playbackScrubber.isPressed)
+    {
+      slider.value = PlaybackScrubber.Progress();
+    }
+  }
+
+  function updateEndTime() {
+    endTime.text = PlaybackScrubber.EndTimeAsString();
+    maxTime.text = qsTr("/ ") + PlaybackScrubber.EndTimeAsString();
+  }
+
+  function updateStartTime() {
+    startTime.text = PlaybackScrubber.StartTimeAsString();
+  }
+
+  function updateCurrentTime() {
```

Comment:
Could you document the functions and the properties? See an example [here](https://github.com/ignitionrobotics/ign-gazebo/blob/master/src/gui/plugins/component_inspector/ComponentInspector.qml).

Reply:
39ca2b20e1f33e08d173513b83b827e692b73f82